src/lib/modules/specialconfigs/index.js

```diff
@@ -162,25 +185,101 @@ class SpecialConfigs {
 
     self.embark.registerActionForEvent("deploy:contract:shouldDeploy", (params, cb) => {
       let cmd = params.contract.deployIf;
+      const contract = params.contract;
       if (!cmd) {
         return cb(params);
       }
 
-      self.events.request('runcode:eval', cmd, (err, result) => {
+      if (typeof cmd === 'function') {
+        this.getOnDeployLifecycleHookDependencies(contract).then(dependencies => {
+          const callbackFn = (value) => {
+          };
```

Comment:
cosmetic: can be one line

Reply:
This one is actually a left over that has been removed in the meantime :)